minijinja/src/value/mod.rs

```diff
@@ -593,13 +606,41 @@ impl Value {
     /// ```
     pub fn as_slice(&self) -> Result<&[Value], Error> {
         match self.0 {
-            ValueRepr::Undefined | ValueRepr::None => Ok(&[][..]),
-            ValueRepr::Seq(ref v) => Ok(&v[..]),
-            _ => Err(Error::new(
-                ErrorKind::InvalidOperation,
-                format!("value of type {} is not a sequence", self.kind()),
-            )),
+            ValueRepr::Undefined | ValueRepr::None => return Ok(&[][..]),
+            ValueRepr::Seq(ref v) => return Ok(&v[..]),
+            ValueRepr::Dynamic(ref dy) if matches!(dy.behavior(), ObjectBehavior::Seq(_)) => {
+                return Err(Error::new(
+                    ErrorKind::InvalidOperation,
+                    "dynamic sequence value cannot be borrowed as slice",
+                ));
+            }
+            _ => {}
```

Comment:
This is pretty surprising. I would advocate for returning a cheap `Slice` object that works for all `Seq`. Or perhaps that's just an `&dyn AsSeq`. If this method is desired, then perhaps adding another `as_seq()` method is advisable.

Reply:
Yes, I want to find a better solution here. One option is obviously to take the entire method away.